File: top-interview-questions/easy/array/two-sum.py
#### https://leetcode.com/explore/interview/card/top-interview-questions-easy/92/array/546/
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen = {}
        for index, number in enumerate(nums):
            if number in seen and seen[number] != index:
                i = seen[number]
                j = index
                return [i, j]
            else:
                seen[target-number] = index

File: 75-blind/arrays/two-sum.go
// https://leetcode.com/problems/two-sum/

package main

func twoSum(nums []int, target int) []int {
	partials := map[int]int{}
	for index, num := range nums {
		p, ok := partials[num]
		if ok {
			return []int{p, index}
		} else {
			partialSum := target - num
			partials[partialSum] = index
		}
	}
	return []int{}
}
